docs: update watch API doc for clarity

yichengq opened the pull request:
follows from #2777 

I added the correct watch command, and changed some sentences to make it more readable.
I also ran the example and correct the commands.

/cc @xiang90 @robszumski @schmichael @ravigadde

robszumski commented on Documentation/api.md:
```diff
@@ -287,26 +287,38 @@ curl 'http://127.0.0.1:2379/v2/keys/foo?wait=true&waitIndex=7'
 If we miss all the 1000 events, we need to recover the current state of the 
-watching key space. First, We do a get and then start to watch from the (etcdIndex + 1).
+watching key space through a get and then start to watch from the
+`X-Etcd-Index + 1`.
```
Is the value of this actually `X-Etcd-Index + 1` or is it `X-Etcd-Index` + 1?

yichengq replied:
X-Etcd-Index is a int variable, and we use the value one bigger than it as the wait index. Would you suggest me to use `X-Etcd-Index` + 1?

robszumski replied:
I think it's slightly more clear if you do `X-Etcd-Index` + 1 but that's just me. I like using backticks for actual specific values that you would see in a log or type into a command line.

robszumski commented on Documentation/api.md:
```diff
@@ -316,23 +328,32 @@ curl 'http://127.0.0.1:2379/v2/keys/foo' -vv
+
+Using the `modifiedIndex + 1` is functionally equivalent for subsequent
+watches, but since it is smaller than the `X-Etcd-Index + 1` we may receive a
+`401` error immediately.
```
Explain what

ravigadde commented on Documentation/api.md:
```diff
@@ -316,23 +328,32 @@ curl 'http://127.0.0.1:2379/v2/keys/foo' -vv
+Unlike watches we use the `X-Etcd-Index + 1` of the response as a `waitIndex`
+instead of the node's `modifiedIndex + 1` for two reasons:
+
+1. The `X-Etcd-Index` will be greater than or equal to the `modifiedIndex`
```
This statement is not accurate? I see watches return with X-Etcd-Index less than modifiedIndex. I set the next watch to greater of the two indicies + 1. This is exactly the question I was asking in #3007

yichengq replied:
This is true for get request because the header and the event returns at the same time. I will clarify this point more.

yichengq commented:
@robszumski @ravigadde Addressed your comments. PTAL

@ravigadde I also clarify X-Etcd-Index more [here](https://github.com/yichengq/etcd/commit/9a61fddd74e6219448c8c54ae4ef5e7043a627f2).

robszumski commented on Documentation/api.md:
````diff
@@ -78,7 +78,7 @@ X-Raft-Index: 5398
 ```
 
-- `X-Etcd-Index` is the current etcd index as explained above.
+- `X-Etcd-Index` is the current etcd index as explained above. Especially, when request is a watch on key space, `X-Etcd-Index` is the current etcd index when watch starts, which means that the watched event may happen after `X-Etcd-Index`.
````
You can remove "especially" here. "When a request is..."

robszumski replied:
"...when **the** watch..."

schmichael commented on Documentation/api.md:
```diff
@@ -316,23 +328,34 @@ curl 'http://127.0.0.1:2379/v2/keys/foo' -vv
+
+1. The `X-Etcd-Index` is always greater than or equal to the `modifiedIndex` when
+   getting key because `X-Etcd-Index` is the current etcd index, and the modification
+   is a history event of etcd store.
```
Grammar is a bit off here...

> when getting key because

to:

> when getting *a* key because

--

> the modification is a history event of etcd store.

to

> the `modifiedIndex` is the index of an event already stored in etcd.

--

Just suggestions. It's perfectly understandable as is.

schmichael commented on Documentation/api.md:
```diff
@@ -316,23 +328,34 @@ curl 'http://127.0.0.1:2379/v2/keys/foo' -vv
+1. The `X-Etcd-Index` is always greater than or equal to the `modifiedIndex` when
+   getting key because `X-Etcd-Index` is the current etcd index, and the modification
+   is a history event of etcd store.
+2. None of those indexes between `modifiedIndex` and `X-Etcd-Index` will be
```
> None of those

to

> None of the

--

Or maybe this sentence instead?

> None of the events represented by indexes between `modifiedIndex` and `X-Etcd-Index` will be related to the key being fetched.

schmichael commented:
LGTM FWIW. Sorry for not being able to follow through with my original PR!

yichengq commented:
@robszumski @schmichael Thanks for reviewing! Changed as you suggest.

schmichael commented:
:+1:

xiang90 commented:
Can we squash the commits and get it merged?

yichengq commented:
Thanks for reviewing! merging this because this is only related to doc and has been read through several times by multiple reviewers.
